From a8d43ec1c7f5448a7f63d9bbd9062d56ee1c7c58 Mon Sep 17 00:00:00 2001
From: Albert Astals Cid <aacid@kde.org>
Date: Wed, 14 Apr 2010 18:21:33 +0000
Subject: update XOPEN_SOURCE to 600 in non standard compile options

it seems helps compiling on some BSD
See also https://bugs.freedesktop.org/show_bug.cgi?id=27610
and http://bugs.gentoo.org/show_bug.cgi?id=314925

--- poppler-0.12.4.orig/cmake/modules/PopplerMacros.cmake
+++ poppler-0.12.4/cmake/modules/PopplerMacros.cmake
@@ -88,7 +88,7 @@
 if(CMAKE_COMPILER_IS_GNUCXX)
   set(CMAKE_C_FLAGS "-Wall -Wno-write-strings -Wno-long-long -Wundef -Wcast-align -Wconversion -Wchar-subscripts -Wall -W -Wpointer-arith -Wwrite-strings -Wformat-security -Wmissing-format-attribute -fno-common ${CMAKE_C_FLAGS}")
   set(CMAKE_CXX_FLAGS "-Wall -Woverloaded-virtual -Wno-write-strings -Wnon-virtual-dtor -Wno-long-long -Wundef -ansi -Wcast-align -Wconversion -Wchar-subscripts -Wall -W -Wpointer-arith -Wwrite-strings -Wformat-security -Wmissing-format-attribute -fno-check-new -fno-common ${CMAKE_CXX_FLAGS}")
-  add_definitions(-D_XOPEN_SOURCE=500 -D_BSD_SOURCE)
+  add_definitions(-D_XOPEN_SOURCE=600 -D_BSD_SOURCE)
   set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "-O2 -g")
   set(CMAKE_CXX_FLAGS_RELEASE        "-O2 -DNDEBUG")
   set(CMAKE_CXX_FLAGS_DEBUG          "-g -O2 -fno-reorder-blocks -fno-schedule-insns -fno-inline")
--- poppler-0.12.4.orig/configure.ac
+++ poppler-0.12.4/configure.ac
@@ -512,7 +512,7 @@
   no)	;;
   yes)	CXXFLAGS="-Wall -Wno-write-strings -Woverloaded-virtual $CXXFLAGS" ;;
   kde)	CXXFLAGS="-Wnon-virtual-dtor -Wno-long-long -Wundef -ansi \
-		-D_XOPEN_SOURCE=500 -D_BSD_SOURCE -Wcast-align \
+		-D_XOPEN_SOURCE=600 -D_BSD_SOURCE -Wcast-align \
 		-Wconversion -Wchar-subscripts -Wall -W -Wpointer-arith \
 		-Wwrite-strings -O2 -Wformat-security \
 		-Wmissing-format-attribute -fno-exceptions -fno-check-new \
